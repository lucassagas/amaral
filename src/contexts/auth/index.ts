export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
