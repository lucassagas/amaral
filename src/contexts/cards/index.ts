export { CardContext, CardContextProvider } from './cards';

