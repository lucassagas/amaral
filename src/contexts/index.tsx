import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
