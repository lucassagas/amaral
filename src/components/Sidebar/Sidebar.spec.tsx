import { AppProvider } from '@/contexts/index'
import { render } from '@testing-library/react'
import { Sidebar } from './Sidebar'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

describe('<Sidebar/>', () => {
  it('should be render correctly', () => {
    const { container } = render(
      <AppProvider>
        <Sidebar />
      </AppProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
