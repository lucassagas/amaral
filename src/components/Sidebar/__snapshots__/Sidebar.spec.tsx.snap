// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> should be render correctly 1`] = `
<div>
  <aside
    class="css-1hjbzlu"
  >
    <div
      class="chakra-stack css-1yucvmi"
    >
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-1azq1gc"
        >
          Geral
        </p>
        <div
          class="chakra-stack css-1vcm6cv"
        >
          <a
            align="center"
            class="chakra-link css-6qooy7"
            href="/app/dashboard"
          >
            <svg
              class="chakra-icon css-1cv0b1u"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M13 21V11h8v10h-8zM3 13V3h8v10H3zm6-2V5H5v6h4zM3 21v-6h8v6H3zm2-2h4v-2H5v2zm10 0h4v-6h-4v6zM13 3h8v6h-8V3zm2 2v2h4V5h-4z"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-y7ioq2"
            >
              Dashboard
            </p>
          </a>
          <a
            align="center"
            class="chakra-link css-6qooy7"
            href="/app/products"
          >
            <svg
              class="chakra-icon css-1cv0b1u"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M6.5 2h11a1 1 0 0 1 .8.4L21 6v15a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V6l2.7-3.6a1 1 0 0 1 .8-.4zM19 8H5v12h14V8zm-.5-2L17 4H7L5.5 6h13zM9 10v2a3 3 0 0 0 6 0v-2h2v2a5 5 0 0 1-10 0v-2h2z"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-y7ioq2"
            >
              Produtos
            </p>
          </a>
        </div>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-1azq1gc"
        >
          Admin
        </p>
        <div
          class="chakra-stack css-1vcm6cv"
        >
          <a
            align="center"
            class="chakra-link css-6qooy7"
            href="/users"
          >
            <svg
              class="chakra-icon css-1cv0b1u"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 7h5v2h-5V7zm-2 5h7v2h-7v-2zm3 5h4v2h-4v-2zM2 22a8 8 0 1 1 16 0h-2a6 6 0 1 0-12 0H2zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4z"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-y7ioq2"
            >
              Usu√°rios
            </p>
          </a>
        </div>
      </div>
    </div>
  </aside>
</div>
`;
